test/bugs-reported/bug4.future

```diff
@@ -0,0 +1,28 @@
+bug: The List Module does not support accessing elements of a list based on index. I feel that this should be supported.
```

Comment:
Yikes! This one's embarassing...

I'd advocate for splitting this up into two smaller programs/futures.

I don't think we want to allow indexing through the ".these" iterator, but we definitely shouldn't emit an internal compiler error.